cibuildwheel/resources/constraints.in

```diff
@@ -0,0 +1,5 @@
+pip
+setuptools
+wheel
+delocate
+virtualenv
```

Comment:
Why no pinned version? Dependabot is wrong? Bump multiple package in one moment can create problem to determine which one are problematic. 

Or it will bump compiled version?

Reply:
From one hand dependabot is noisy. It create PR for each requirements bump. For other hand. you do not need to do this work manually. You see new PR and see result of the test if this release break something. 

I do not have experience with keeping requirements for python 2.7. 

But maybe https://requires.io can be used to and https://libraries.io/ can be used to trace new releases.

Reply:
> Why no pinned version?

constraints.txt contains the pinned versions. If we have an incompatibility (like the recent `virtualenv<20` issue) that would be here. Otherwise, we want `bin/update-constraints` to choose the latest, so we don't pin this file.

> Dependabot is wrong? Bump multiple package in one moment can create problem to determine which one are problematic.

Dependabot won't bump separate constraints for python2 and python3, and won't update the docker images. Requires.io doesn't seem to support it either.

Currently, I'm thinking I'd bump these immediately before a release, but it could be good to have an automated (nightly) build job that checks for new versions, then run tests and merges if they pass. But we can consider that if bumping becomes a big maintenance burden.

It's not essential that cibuildwheel is always bleeding edge - the wheel standard is very stable.

Reply:
>  but it could be good to have an automated (nightly) build job that checks for new versions, then run tests and merges if they pass.

That's some extra convenience tooling for us. Let's not have that stop us from finishing this PR.

Reply:
> It's not essential IMO that cibuildwheel is always bleeding edge - the wheel standard is very stable.

And I more or less agree, though we háve been waiting on a `wheel` release, recently :-D